synapse/rest/client/v1/register.py

```diff
@@ -410,12 +410,14 @@ def _do_create(self, user_json):
             raise SynapseError(400, "Failed to parse 'duration_seconds'")
         if duration_seconds > self.direct_user_creation_max_duration:
             duration_seconds = self.direct_user_creation_max_duration
+        password_hash = user_json["password_hash"].encode("utf-8") if user_json["password_hash"] else None
```

Comment:
You probably want to do:

```python
... if user_json.get("password_hash") else None
```

So that it doesn't explode if no `password_hash` key is present.